code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/AEPMessage.java

```diff
@@ -372,13 +372,18 @@ void showInRootViewGroup() {
     /** Tears down views and listeners used to display the {@link AEPMessage}. */
     void cleanup() {
         Log.trace(ServiceConstants.LOG_TAG, TAG, "Cleaning the AEPMessage.");
-        delegateFullscreenMessageDismiss();
+        if (!messagesMonitor.dismiss()) {
+            return;
+        }
 
-        if (messagesMonitor != null) {
-            messagesMonitor.dismissed();
+        // notify message listeners
+        listener.onDismiss(this);
```

Comment:
if (isVisible) { // If this flag is false, it means we had some error and did not call onShow() notifiers
    listener.onDismiss(this);

    MessagingDelegate delegate = ServiceProvider.getInstance().getMessageDelegate()
    if(delegate != null) {
     delegate.onDismiss(this);
   }
}